### PR TITLE
topology: add TGL tplg for max98357a amp and ALC5682 Headset codec

### PR DESCRIPTION
**Pull request by sathya-nujella**
Add TGL topology for the DUT which has the following audio configuration:
        Max98357A speaker (I2S)
        ALC5682 headset codec (I2S)
        DMIC
        4 HDMI devices

### CODE REVIEW

**Comment by sathya-nujella** [comment]
Hi @ranj063,
Removed line spacing and updated the patch. Please review.

File: tools/topology/sof-tgl-max98357a-rt5682.m4 [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+include(`platform/intel/tgl.m4')
+include(`platform/intel/dmic.m4')
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM0 <---> volume <----> SSP1  (Speaker - max98357a)
+# PCM1 <---> volume <----> SSP0  (Headset - ALC5682)
+# PCM99 ----> volume -----> DMIC01 (dmic0 capture)
```
Comment (lyakh) [inline tools/topology/sof-tgl-max98357a-rt5682.m4]: I think arrows should be pointing in the opposite direction

File: tools/topology/sof-tgl-max98357a-rt5682.m4 [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+# Include Tigerlake DSP configuration
+include(`platform/intel/tgl.m4')
+include(`platform/intel/dmic.m4')
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM0 <---> volume <----> SSP1  (Speaker - max98357a)
```
Comment (lyakh) [inline tools/topology/sof-tgl-max98357a-rt5682.m4]: this seems to be playback only, not duplex, so arrows one direction, please

File: tools/topology/sof-tgl-max98357a-rt5682.m4 (hunk 1 above)
Comment (sathya-nujella) [reply tools/topology/sof-tgl-max98357a-rt5682.m4]: fixed and updated patch now.

File: tools/topology/sof-tgl-max98357a-rt5682.m4 (hunk 2 above)
Comment (sathya-nujella) [reply tools/topology/sof-tgl-max98357a-rt5682.m4]: fixed and updated patch now.

**Comment by sathya-nujella** [comment]
> comments inline

Thank you.
fixed and updated patch now.
Also fixed check-patch warnings.

**Comment by sathya-nujella** [comment]
> For me looks ok, but I'd wait for @plbossart or @ranj063 review, to have a look from a driver side.

Hi @ranj063 , @plbossart 
Once this BUG: https://github.com/thesofproject/sof/issues/2084 is fixed.. we will update if any changes needed for this m4 file.  Also if we have to avoid any M/N settings after fix lands, we can add one more patch for this m4 file.
For now, this is the working m4 file on DUT. so, asking for your review of this one.

**Comment by ranj063** [comment]
@sathya-nujella could you please aliign the dmic Dec name with PR 2186?

**Comment by sathya-nujella** [comment]
> @sathya-nujella could you please aliign the dmic Dec name with PR 2186?

Changed PCM name now and updated patch.

Also cross verified in DUT, reflected the change.
**Before:**
localhost /home/root # arecord -l
..
card 0: sofsofrt5682 [sof-sof_rt5682], device 99: **DMIC01** (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**After:**
localhost ~ # arecord -l
..
card 0: sofsofrt5682 [sof-sof_rt5682], device 99: **DMIC48kHz** (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**Comment by jajanusz** [comment]
@ranj063 is it ok now?

**Pull request merged**